pulsar-common/src/main/java/com/yahoo/pulsar/common/api/Commands.java

```diff
@@ -611,7 +611,6 @@ public static long initBatchMessageMetadata(PulsarApi.MessageMetadata.Builder me
         messageMetadata.setPublishTime(builder.getPublishTime());
         messageMetadata.setProducerName(builder.getProducerName());
         messageMetadata.setSequenceId(builder.getSequenceId());
-        messageMetadata.setReplicatedFrom(builder.getReplicatedFrom());
```

Comment:
Good catch! Would it be worth to use "if (hasReplicatedFrom()) setTeplicatedFrom()" ?